scripts/ssl_tts/make_supdata.py

```diff
@@ -0,0 +1,434 @@
+import argparse
+import json
+import os
+import pickle
+import shutil
+import time
+from email.mime import audio
+from locale import normalize
+from multiprocessing import Pool
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import hydra.utils
+import librosa
+import numpy as np
+import torch
+from omegaconf import open_dict
+from tqdm import tqdm
+
+from nemo.collections.asr.models import ssl_models
+from nemo.collections.asr.parts.preprocessing.features import WaveformFeaturizer
+from nemo.collections.asr.parts.preprocessing.segment import AudioSegment
+from nemo.collections.tts.models import ssl_tts
+from nemo.collections.tts.torch.helpers import get_base_dir
+from nemo.collections.tts.torch.tts_tokenizers import EnglishCharsTokenizer
+from nemo.core.classes import Dataset
+from nemo.utils import logging
+
+SUP_DATA_DIR = None
+
+wav_featurizer = WaveformFeaturizer(sample_rate=22050, int_values=False, augmentor=None)
+
+
+class AudioDataset(Dataset):
```

Comment:
Could this be refactored to build off of the SSLVocoderDataset to reduce code reuse and overhead? It looks like several of the functions in here are similar or identical.

Reply:
Left it like this for now but can potentially be inherited. Just wanted to keep the __init__ arguments simple so we used a plain dataset class.

Reply:
Sure, fine with me 👍